ambry-commons/src/main/java/com/github/ambry/compression/BaseCompression.java

```diff
@@ -432,28 +245,35 @@ public int decompress(ByteBuffer compressedBuffer, ByteBuffer decompressedBuffer
     Utils.checkNotNullOrEmpty(compressedBuffer, "compressedBuffer cannot be null or empty.");
     Utils.checkNotNullOrEmpty(decompressedBuffer, "decompressedBuffer cannot be null or empty.");
 
-    // Save the position.  Use position instead of mark just in case decompressNative() uses mark().
+    // Check compressed buffer size - must be bigger than header size without algorithm name.
+    if (compressedBuffer.remaining() < VERSION_AND_ORIGINAL_SIZE_SIZE + ALGORITHM_NAME_LENGTH_SIZE) {
```

Comment:
This math appears often together. Introduce a new constant as 

MIN_COMPRESSED_BUFFER_BYTES = 
VERSION_AND_ORIGINAL_SIZE_SIZE + ALGORITHM_NAME_LENGTH_SIZE

Reply:
ok, the ending SIZE_SIZE is a bit confusing. I'll let it be.